frontend/src/components/input.tsx

```diff
@@ -36,23 +55,24 @@ export const Input = ({ sendMessage, waiting }: InputProps) => {
   return (
     <>
       <form onSubmit={onSend} className={styles.inputContainer}>
-        <input
-          className={styles.input}
-          onChange={onChange}
-          onSubmit={onSend}
+        <textarea
+          className={styles.textarea}
+          ref={textareaRef}
           placeholder="Send a Message..."
-          type="text"
           value={userInput}
+          onChange={onChange}
+          rows={1}
         />
-        <button
-          className={classNames(styles.sendButton, {
-            [styles.disabled]: sendDisabled,
-          })}
-          onClick={onSend}
+        <div className={styles.uploadButton_container}>
+          <button className={styles.uploadButton} disabled={sendDisabled}>
+            <img src={UploadIcon} />
+          </button>
+        </div>
+        <Button
+          icon={RightArrowIcon}
           disabled={sendDisabled}
-        >
-          <img src={RightArrow} />
-        </button>
+          onClick={() => onSend}
```

Comment:
This will just be a function that returns the `onSend` function rather than calls it. The send still works as the button defaults to submitting the form as well (which also calls onSend).

We probably want to either fix the onClick types so can use onSend here, or make it explicit we're not using the onClick by making it optional?